### PR TITLE
LPS-134690 Categories section to be ordered into vocabularies

### PR DESCRIPTION
**Pull request by beltranrengifo**
[See original PR](https://github.com/brianchandotcom/liferay-portal/pull/112720)
[Jira ticket](https://issues.liferay.com/browse/LPS-134690)

## Motivation
This PR addresses an ordering problem in the rendering of the vocabularies for each Content Dashboard Item in the Info Panel.

## What has been done
We group and sort global vocabularies, then _non globals_

#### Result
image

## How to test
Please follow this video demo
https://drive.google.com/file/d/10HRXKijTMWyO8UNoIo4dWZnunXbDMjuF/view?usp=sharing

Or

-  Create two sites and an Asset Library
- Create some vocabularies and categories for each site/library
- Add some categories to default global categories, and add some new global vocabularies
- Create at least one web content for each site/library and assign all categories you want for each content
- Navigate to the Content Dashboard and open the info panel using the info icon or the actions dropdown

#### Expected
The vocabularies are grouped: first the global ones, then the rest
The vocabularies are sorted alphabetically

### CODE REVIEW

**Comment by beltranrengifo** [comment]
cI:test:sf

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:test:sf

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:test:sf

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/utils/renderVocabularies.js [hunk 1]
```diff
@@ -21,22 +29,44 @@ const getCategoriesCountFromVocabularies = (vocabularies) =>
...
 	);
+
+/**
+ * Sorts an array by a given criteria, being the value of this criteria a string
+ * If no key is present the sorting applies directly over each item, assuming they are strings
+ * @param {array} array A collection of vocabularies from a Content Dashboard Item.
+ * @param {string} [key = ''] A string representing the property
+ * @returns {array} An array sorted by a property
+ */
+const sortByStrings = ({array, key = ''}) => {
```
[the remark this answers is not in the document]
Comment (claraizquierdo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/utils/renderVocabularies.js]: I guess we use different code styles xD Anyway, if it receives an object I think you should change it in the comment in lines 57 and 58.

[the remark this answers is not in the document]
Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/utils/renderVocabularies.js]: I try to be flexible in this, not a rigid rule, but generally I think is safer.

[the remark this answers is not in the document]
Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelInfoView/utils/renderVocabularies.js]: Pushed the docs update, thxs!

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Review by claraizquierdo — APPROVED** [review]

**Comment by beltranrengifo** [comment]
ci:forward

**Pull request closed without merging**